hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/TinkerpopBackend.java

```diff
@@ -178,7 +179,7 @@ public Element find(CanonicalPath element) throws ElementNotFoundException {
                     });
         }
 
-        return new Page<>(q2.toList(), pager, q.getCount("total"));
+        return new Page<>(q2, pager, q.getCount("total"));
```

Comment:
I'll try something like passing the reference to `q` into the page instance itself and setting the `totalSize` (by calling `q.getCount("total")`) once the `hasNext()` returns false.

However this smells like a memory leak (holding the reference to the pipe in a page instance), perhaps the weak reference would do the trick here.